docs/index.rst

```diff
@@ -4,4 +4,315 @@ JDBC Connector
+
+Now, run the connector in a standalone Kafka Connect worker in another terminal (this assumes
+Avro settings and that Kafka and the Schema Registry are running locally on the default ports)::
+
+   $ ./bin/connect-standalone ./config/connect-standalone.properties quickstart-sqlite.properties
+
+You should see the process start up and log some messages, and then it will begin executing
+queries and sending the results to Kafka. In order to check that it has copied the data that was
+present when we started Kafka Connect, start a console consumer, reading from the beginning of
+the topic::
+
+   $ ./bin/kafka-avro-console-consumer --new-consumer --bootstrap-server localhost:9092 --topic test-sqlite-jdbc-accounts --from-beginning
+   {"id":1,"name":{"string":"alice"}}
+   {"id":2,"name":{"string":"bob"}}
+
+The output shows the two records as expected, in the JSON encoding of the Avro records. Now,
+keeping that running, add another record via the SQLite command prompt::
+
+   sqlite> INSERT INTO accounts(name) VALUES('cathy');
+
+If you switch back to the console consumer, you should see the new record added (and,
+importantly, the old entries are not repeated)::
+
+   {"id":3,"name":{"string":"cathy"}}
+
+Note that the default polling interval is 5 seconds, so it may take a few seconds to show up.
+Depending on your expected rate up updates or desired latency, a smaller poll interval could be
+used to deliver updates more quickly.
+
+Of course, all the normal features of Kafka Connect, including offset management and fault
+tolerance, work with the JDBC connector. You can restart and kill the processes and they will
+pick up where they left off, copying only new data (as defined by the ``mode`` setting).
+
+Features
+--------
+
+The JDBC connector supports copying tables with a variety of JDBC data types, adding and removing
+tables from the database dynamically, whitelists and blacklists, varying polling intervals, and
+other settings. However, the most important features for most users are the settings controlling
+how data is incrementally copied from the database.
+
+Mapping to Kafka Connect Partitioned Streams & Offsets
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Kafka Connect uses `partitioned streams` as its core data model. In order to map to Kafka Connect
+records, the JDBC connector must model its data as a set of partitions, each with an ordered
+sequence of updates with unique offsets. In the case of a process restart or failure, these
+offsets are used to resume reading from the last position that the system can guarantee has been
+processed.
+
+To map to this model, the JDBC connector must map partitions and offsets. Partitions is
+handled on a per-table basis: each table represents a single partition and every insertion or
+update of a row corresponds to a single record in that partition. As a special case, custom
+queries (which disable table-based processing) are treated as a single-partition stream.
```

Comment:
Makes sense. Fixed to just give a rough intro explaining how the incremental querying works and then cleaned up the explanations of the different modes a bit to try to keep them as simple as possible.